peripheral/eth.c

```diff
@@ -41,15 +41,10 @@
 #include <sys/types.h>
 #include <fcntl.h>
 
-#include <sys/poll.h>
+#include <poll.h>
 #include <unistd.h>
 #include <errno.h>
 
-#if HAVE_LINUX_IF_TUN_H==1
-#include <linux/if.h>
-#include <linux/if_tun.h>
-#endif
-
```

Comment:
What was the thinking about removing this? If seems to be conditional anyway?

Reply:
I didn't remove it. I moved it to later in the file.

Reply:
Ah OK - missed that. What was the thinking behind the move? General structure of the source code is system headers then user headers.

Reply:
The issue is that some of those other user headers include system headers as well. And Linux headers have to be included after libc headers, due to how `linux/libc-compat.h` works. Otherwise I get:
```
In file included from /usr/include/net/ethernet.h:10,
                 from ../../peripheral/eth.h:31,
                 from ../../peripheral/eth.c:57:
/usr/include/netinet/if_ether.h:114:8: error: redefinition of 'struct ethhdr'
  114 | struct ethhdr {
      |        ^~~~~~
In file included from /usr/include/linux/if_tun.h:21,
                 from ../../peripheral/eth.c:50:
/usr/include/linux/if_ether.h:169:8: note: originally defined here
  169 | struct ethhdr {
      |        ^~~~~~
cc1: all warnings being treated as errors
```